configTools/src/test/scala/com/gu/janus/policy/IamTest.scala

```diff
@@ -0,0 +1,374 @@
+                       |    "*"
+                       |  ]
+                       |}""".stripMargin
+
+      basicStatement.asJson.spaces2 shouldBe expected
+    }
+
+    "should encode statement with ID correctly" in {
+      val statementWithId = basicStatement.copy(id = Some("statement1"))
+      val expected = """{
+                       |  "Effect" : "Allow",
+                       |  "Action" : [
+                       |    "iam:PutRolePolicy",
+                       |    "iam:getRole"
+                       |  ],
+                       |  "Resource" : [
+                       |    "*"
+                       |  ],
+                       |  "Sid" : "statement1"
+                       |}""".stripMargin
+
+      statementWithId.asJson.spaces2 shouldBe expected
+    }
+
+    "should encode statement with single condition correctly" in {
+      val statementWithCondition = basicStatement.copy(
+        conditions = List(
+          Condition("aws:SourceIp", "StringEquals", List("203.0.113.0/24"))
+        )
+      )
+
+      val expected = """{
+                       |  "Effect" : "Allow",
+                       |  "Action" : [
+                       |    "iam:PutRolePolicy",
+                       |    "iam:getRole"
+                       |  ],
+                       |  "Resource" : [
+                       |    "*"
+                       |  ],
+                       |  "Condition" : {
+                       |    "StringEquals" : {
+                       |      "aws:SourceIp" : [
+                       |        "203.0.113.0/24"
+                       |      ]
+                       |    }
+                       |  }
+                       |}""".stripMargin
+
+      statementWithCondition.asJson.spaces2 shouldBe expected
+    }
+
+    "should encode statement with single condition with multiple values correctly" in {
+      val statementWithCondition = basicStatement.copy(
+        conditions = List(
+          Condition(
+            "aws:SourceIp",
+            "StringEquals",
+            List("203.0.113.0/24", "203.0.113.1/24")
+          )
+        )
+      )
+
+      val expected = """{
+                       |  "Effect" : "Allow",
+                       |  "Action" : [
+                       |    "iam:PutRolePolicy",
+                       |    "iam:getRole"
+                       |  ],
+                       |  "Resource" : [
+                       |    "*"
+                       |  ],
+                       |  "Condition" : {
+                       |    "StringEquals" : {
+                       |      "aws:SourceIp" : [
+                       |        "203.0.113.0/24",
+                       |        "203.0.113.1/24"
+                       |      ]
+                       |    }
+                       |  }
+                       |}""".stripMargin
+
+      statementWithCondition.asJson.spaces2 shouldBe expected
+    }
+
+    "should encode statement with multiple conditions correctly" in {
```

Comment:
Do we need to test the case where we have multiple conditions of the same type, to check it merges properly?

```scala
conditions = List(
          Condition("aws:SourceIp", "StringEquals", List("203.0.113.0/24")),
          Condition(
            "aws:TokenIssueTime",
            "StringEquals", // <-- THE SAME HERE
            List("2025-02-27T09:33:01.000Z")
          )
        )
```

Reply:
Done in 7fbe8f459213feef1410a72b35feed80c4860b38